CONTRIBUTING.md

```diff
@@ -64,28 +66,23 @@ We use [flake8](https://pypi.org/project/flake8/) for quick style checks and [bl
 ### Unit Tests
 
 Tests are written using [pytest](https://github.com/pytest-dev/pytest) and executed together with [codecov](https://github.com/codecov/codecov-action) for coverage reports.
-To perform the tests, execute the following (make sure pytest is installed):
+We use [tox](https://tox.wiki/en/latest/) for test automation. 
 ```bash
-pytest
+# To list all test environments, run:
+python3 -m tox list
+# To execute all test environments run:
+python3 -m tox run
+# Or, e.g., you can run only tests for pytho3.8 with:
+python3 -m tox run -e py38
+# Or, e.g., run type checking with:
+python3 -m tox run -e type
+# If you want to pass CLI arguments to pytest, add them after --, e.g., you might want to split execution between 4 cpu cores, 
+# show verbose logging, and run only robustness tests.
+python3 -m tox run py310 -- -n 4 -s -v -m "robustness"
+# You still can run tests using your active local interpreter by:
+python -m pytest 
 ```
 
-
-... optionally, you could split test execution between multiple CPU cores using [pytest-xdist](https://github.com/pytest-dev/pytest-xdist)
-```bash
-pytest tests -n auto
-```
-
-... alternatively, to get additionaly coverage details, run:
-```bash
-pytest --cov=. --cov-report term-missing
```

Comment:
What is the tox equivalent of generating cov report? Can we add this above.

Reply:
What do you mean by equivalent? The coverage report will still be generated using `pytest-cov` as before. 
I have just taken the command we used to generate coverage report in GitHub action, and put it as a command for tox environment called coverage. You can find those in `tox.ini`, on line 43.  To generate a coverage report, you'd need to run `tox run -e coverage`.

Reply:
> What is the tox equivalent of generating cov report? Can we add this above.

What do you mean by equivalent? The coverage report will still be generated using `pytest-cov` as before. 
I have just taken the command we used to generate coverage report in GitHub action, and put it as a command for tox environment called coverage. You can find those in `tox.ini`, on line 43.  To generate a coverage report, you'd need to run `tox run -e coverage`.

Reply:
Great, thanks for clarification. Can you add that comment here: `tox run -e coverage`.

Reply:
please add the tox coverage command

Reply:
done